aiidalab_widgets_base/viewers.py

```diff
@@ -554,6 +555,59 @@ def _observe_all_representations(self, change):
         if change["new"]:
             self._all_representations[-1].viewer_class = self
 
+    def _compute_bonds(self, structure, radius=1.0, color="element", povray=False):
+        """Create an array of bonds for the structure."""
+
+        import ase.neighborlist
+
+        radius = radius / 5
+
+        cutoff = ase.neighborlist.natural_cutoffs(structure, mult=1.09)
+        bonds = []
+
+        def povray_cylinder(v1, v2, radius, atom, color):
+            color = np.array(Colors[atom.symbol])
+            return vapory.Cylinder(
+                v1,
+                v2,
+                radius,
+                vapory.Pigment("color", np.array(color)),
+                vapory.Finish("phong", 0.8, "reflection", 0.05),
+            )
+
+        def cylinder(v1, v2, radius, atom, color):
+            color = (
+                colors.jmol_colors[atom.number].tolist()
+                if color == "element"
+                else RGB_colors[color]
+            )
+            return (
+                "cylinder",
+                tuple(v1),
+                tuple(v2),
+                tuple(color),
+                radius,
+            )
+
+        # Choose the correct function for computing the cylinder.
+        compute_cylinder = povray_cylinder if povray else cylinder
+
+        if len(structure) > 1:
+            ii, jj, bond_lengths = ase.neighborlist.neighbor_list(
```

Comment:
It's a distance vector, not the length.
```suggestion
            ii, jj, bond_vectors = ase.neighborlist.neighbor_list(
```

Reply:
Thanks, changed in https://github.com/aiidalab/aiidalab-widgets-base/pull/535/commits/6ff62b473b3227557d6e6549eb5193ee7bafb7c7